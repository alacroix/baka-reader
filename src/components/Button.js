// @flow

import React from 'react';
import { TouchableOpacity, Text, StyleSheet, View } from 'react-native';

import appStyle from 'bakareader/src/appStyle';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    minHeight: appStyle.dimensions.touchableHeight,
    minWidth: 64,
    paddingHorizontal: appStyle.grid.x1,
  },
  button: {
    justifyContent: 'center',
    height: appStyle.dimensions.visibleButtonHeight,
    paddingHorizontal: appStyle.grid.x1,
  },
  text: {
    textAlign: 'center',
    fontSize: appStyle.font.size.default,
  },
  raisedButton: {
    backgroundColor: appStyle.colors.primary,
  },
  raisedText: {
    color: appStyle.colors.lightText,
  },
  flatButton: {
    backgroundColor: 'transparent',
  },
  flatText: {
    color: appStyle.colors.primary,
  },
});

type PropsType = {
  children: string,
  type?: 'flat' | 'raised',
  onPress?: () => void,
}

function Button(props: PropsType) {
  switch (props.type) {
    case 'flat':
      return (
        <TouchableOpacity activeOpacity={0.6} onPress={props.onPress} style={styles.container}>
          <View style={[styles.button, styles.flatButton]}>
            <Text style={[styles.text, styles.flatText]}>{props.children.toUpperCase()}</Text>
          </View>
        </TouchableOpacity>
      );
    default:
      return (
        <TouchableOpacity activeOpacity={0.6} onPress={props.onPress} style={styles.container}>
          <View style={[styles.button, styles.raisedButton]}>
            <Text style={[styles.text, styles.raisedText]}>{props.children.toUpperCase()}</Text>
          </View>
        </TouchableOpacity>
      );
  }
}

Button.defaultProps = {
  type: 'raised',
  onPress: () => {},
};

export default Button;
