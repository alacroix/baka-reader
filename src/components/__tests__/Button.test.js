import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';

describe('Button', () => {
  it('should render the button', () => {
    const tree = renderer.create(
      <Button>Hello</Button>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
