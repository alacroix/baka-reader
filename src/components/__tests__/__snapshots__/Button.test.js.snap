// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginVertical": 8,
      "minHeight": 48,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#05A5D1",
        "height": 36,
        "justifyContent": "center",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FAFAFA",
            "fontSize": 13,
            "textAlign": "center",
          },
        ]
      }
    >
      HELLO
    </Text>
  </View>
</View>
`;
